### PR TITLE
Update tests for OCaml 5

### PR DESCRIPTION
**Pull request by hhugo**

### CODE REVIEW

File: compiler/tests-check-prim/output [hunk 1]
```diff
@@ -22,6 +30,18 @@ caml_int64_sub_native
...
+caml_ml_condition_broadcast
+caml_ml_condition_new
+caml_ml_condition_signal
+caml_ml_condition_wait
+caml_ml_domain_cpu_relax
+caml_ml_mutex_lock
+caml_ml_mutex_new
+caml_ml_mutex_try_lock
+caml_ml_mutex_unlock
+caml_obj_compare_and_swap
```
Comment (hhugo) [inline compiler/tests-check-prim/output:43]: This should probably be implemented

File: compiler/tests-check-prim/output [hunk 2]
```diff
@@ -22,6 +30,18 @@ caml_int64_sub_native
...
+caml_obj_is_shared
```
Comment (hhugo) [inline compiler/tests-check-prim/output:44]: This should probably be implemented

File: compiler/tests-jsoo/test_marshal.ml [hunk 3]
```diff
@@ -108,12 +108,6 @@ let%expect_test _ =
   in
   let v = Marshal.from_string data 0 in
   let () = assert (1L = v) in
-  [%expect {||}];
-  let data =
-    "\132\149\166\190\000\000\000\012\000\000\000\001\000\000\000\004\000\000\000\003\018_j\000\000\000\000\000\000\000\000\001"
```
Comment (hhugo) [inline compiler/tests-jsoo/test_marshal.ml:113]: Support for Code_custom was dropped

File: runtime/prng.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
+
+//Provides: caml_lxm_next
```
Comment (hhugo) [inline runtime/prng.js:2]: This needs to be tested

Comment (hhugo) [reply runtime/prng.js:2]: done

File: compiler/tests-check-prim/output (hunk 2 above)
Comment (hhugo) [reply compiler/tests-check-prim/output]: done

File: compiler/tests-check-prim/output (hunk 1 above)
Comment (hhugo) [reply compiler/tests-check-prim/output]: done

File: compiler/tests-jsoo/test_marshal.ml (hunk 3 above)
Comment (hhugo) [reply compiler/tests-jsoo/test_marshal.ml:113]: https://github.com/ocaml/ocaml/pull/11223

**Comment by novemberkilo** [comment]
This PR may be relevant #1296

**Pull request closed without merging**